homeassistant/components/sensibo/climate.py

```diff
@@ -95,11 +103,37 @@ async def async_setup_entry(
 
     coordinator: SensiboDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
 
-    async_add_entities(
+    entities = [
         SensiboClimate(coordinator, device_id, hass.config.units.temperature_unit)
         for device_id, device_data in coordinator.data.items()
         # Remove none climate devices
         if device_data["hvac_modes"] and device_data["temp"]
+    ]
+
+    async_add_entities(entities)
+
+    async def async_assume_state(service: ServiceCall) -> None:
+        """Set state according to external service call.."""
+        if entity_ids := service.data.get(ATTR_ENTITY_ID):
+            target_climate = [
+                entity for entity in entities if entity.entity_id in entity_ids
+            ]
+        else:
+            target_climate = entities
+
+        update_tasks = []
+        for climate in target_climate:
+            await climate.async_assume_state(service.data.get(ATTR_STATE))
+            update_tasks.append(climate.async_update_ha_state(True))
+
+        if update_tasks:
+            await asyncio.wait(update_tasks)
+
+    hass.services.async_register(
```

Comment:
In a follow up, this service should preferably be migrated to our entity service helper.